bot.js

```diff
@@ -85,61 +85,57 @@ mongoClient.connect(err => {
-                            foundArr.push(entry.user_login);
-                            onlineStatus[foundIndex].status = entry.type;
-                            twitchEmbed.execute(client, entry, pingRoles, onlineStatus[foundIndex].channelIDs);
-                        } else if (foundIndex >= 0 && onlineStatus[foundIndex].status == entry.type) {
-                            foundArr.push(entry.user_login);
-                        } else if (foundIndex == -1) {
-                            console.log("Stream data received for non followed stream");
-                        }
-                    });
-                    onlineStatus.forEach((element, index) => {
-                        if (!foundArr.includes(element.name))
-                            onlineStatus[index].status = "offline";
-                    });
-                }).catch(err => {
-                    console.log(err.response.data);
-
-                    if (err.response.status == 401) {
-                        getTwitchAccessToken();
-                        setTimeout(checkTwitchChannels, 3000);
+        const pingRoles = await twitchDb.collection('pingRoles').find({}).toArray()
+        if (pingRoles.length > 0) {
+            let twitchGetRequest = `https://api.twitch.tv/helix/streams?user_login=${onlineStatus[0].name}`;
+            for (let i = 1; i < onlineStatus.length; i++) {
+                twitchGetRequest += '&user_login=' + onlineStatus[i].name;
+            }
+            try {
+                const response = await axios.get(twitchGetRequest, {
+                    headers: { 'client-id': config.twitchClientID, Authorization: 'Bearer ' + twitchAccessToken }
+                })
+                console.log(Date(), response.data.data);
+                let foundArr = [];
+                response.data.data.forEach(entry => {
+                    const foundElement = onlineStatus.find(element => element.name == entry.user_login)
+                    if (!foundElement) {
+                        console.log("Stream data received for non followed stream");
+                    } else if (foundElement.status != entry.type) {
+                        foundArr.push(entry.user_login);
+                        foundElement.status = entry.type;
+                        twitchEmbed.execute(client, entry, pingRoles, foundElement.channelIDs);
+                    } else if (foundElement.status == entry.type) {
+                        foundArr.push(entry.user_login);
+                    }
+                });
+                onlineStatus.forEach((element, index) => {
+                    if (!foundArr.includes(element.name)) {
+                        onlineStatus[index].status = "offline";
                     }
                 });
+            } catch (err) {
+                console.log(err.response.data);
+                if (err.response.status == 401) {
+                    await getTwitchAccessToken()
+                      .then(checkTwitchChannels)
```

Comment:
if/when `getTwitchAccessToken` finishes resolving, it'll then retry this same function again, w/o adding a duplicate instance to the the event loop